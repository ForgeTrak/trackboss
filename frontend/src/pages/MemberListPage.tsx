import React from 'react';
import { Box, ChakraProvider } from '@chakra-ui/react';
import theme from '../theme';
import Header from '../components/Header';
import MemberSummaryModal from '../components/MemberSummaryModal';
import { Member } from '../../../src/typedefs/member';
import { Bike } from '../../../src/typedefs/bike';
import MemberList from '../components/MemberList';

const member: Member = {
    memberId: 1,
    membershipAdmin: 'true',
    active: true,
    memberType: 'admin',
    firstName: 'test',
    lastName: 'member',
    phoneNumber: '1234',
    email: 'user@example.com',
    uuid: '',
    occupation: 'secretary',
    birthdate: '',
    dateJoined: '',
    address: '',
    city: '',
    state: '',
    zip: '',
    lastModifiedDate: '',
    lastModifiedBy: '',
};

const memberFamily: Member[] = [
    {
        memberId: 1,
        membershipAdmin: 'true',
        active: true,
        memberType: 'member',
        firstName: 'test',
        lastName: 'member',
        phoneNumber: '1234',
        email: 'user@example.com',
        uuid: '',
        occupation: '',
        birthdate: '',
        dateJoined: '',
        address: '',
        city: '',
        state: '',
        zip: '',
        lastModifiedDate: '',
        lastModifiedBy: '',
    },
    {
        memberId: 2,
        membershipAdmin: 'string',
        active: true,
        memberType: 'member',
        firstName: 'Jon',
        lastName: 'Smith',
        phoneNumber: '1234',
        email: 'user@example.com',
        uuid: '',
        occupation: '',
        birthdate: '',
        dateJoined: '',
        address: '',
        city: '',
        state: '',
        zip: '',
        lastModifiedDate: '',
        lastModifiedBy: '',
    },
];

const memberBikes: Bike[] = [
    {
        bikeId: 0,
        year: '2012',
        make: 'honda',
        model: 'shitbox',
        membershipAdmin: 'string',
    },
];

function MemberListPage() {
    return (
        <ChakraProvider theme={theme}>
            <Header title="Members" activeButtonId={3} />
            <MemberSummaryModal memberInfo={member} memberFamily={memberFamily} memberBikes={memberBikes} />
            <Box pt="3em">
                <MemberList />
            </Box>
        </ChakraProvider>
    );
}

export default MemberListPage;
