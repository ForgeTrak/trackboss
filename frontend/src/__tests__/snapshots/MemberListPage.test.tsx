import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MemberListPage from '../../pages/MemberListPage';

describe('Member List Page', () => {
    it('renders correctly', () => {
        const page = renderer.create(
            <BrowserRouter>
                <MemberListPage />
            </BrowserRouter>
        ).toJSON();
        expect(page).toMatchSnapshot();
    });
});