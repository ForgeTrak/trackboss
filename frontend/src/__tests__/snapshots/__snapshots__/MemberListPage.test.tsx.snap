// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member List Page renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="css-arm2n8"
    >
      <div
        class="chakra-stack css-84zodg"
      >
        <button
          aria-label="Back"
          class="chakra-button css-76fq9n"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="lg"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="lg"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
            />
          </svg>
        </button>
        <div>
          <button
            aria-label="Menu"
            class="chakra-button css-1y85x9m"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="css-17xejub"
      />
      <div
        class="css-0"
      >
        <h2
          class="chakra-heading css-1ht9xq7"
        >
          Members
        </h2>
      </div>
      <div
        class="css-17xejub"
      />
    </div>
  </div>
  <div
    class="css-wxr3l5"
  >
    <div>
      <div
        class="sc-dmRaPn gelpCx"
      >
        <div
          class="sc-fLlhyt ifOHjV"
        >
          <div
            class="sc-bczRLJ eSTlnH rdt_Table"
            role="table"
          >
            <div
              class="sc-gsnTZi idPHNo rdt_TableHead"
              role="rowgroup"
            >
              <div
                class="sc-dkzDqf gZWIKX rdt_TableHeadRow"
                role="row"
              >
                <div
                  class="sc-hKMtZM sc-eCYdqJ sc-iqcoie dAGdUH bnAwAJ crSOFH rdt_TableCol"
                  data-column-id="1"
                >
                  <div
                    class="sc-crXcEl gJhVev rdt_TableCol_Sortable"
                    data-column-id="1"
                    data-sort-id="1"
                    role="columnheader"
                    tabindex="0"
                  >
                    <div
                      class="sc-evZas eUeqdG"
                      data-column-id="1"
                    >
                      Name
                    </div>
                    <span
                      class="sc-kDDrLX fTxtMi"
                    >
                      ▲
                    </span>
                  </div>
                </div>
                <div
                  class="sc-hKMtZM sc-eCYdqJ sc-iqcoie dAGdUH bnAwAJ crSOFH rdt_TableCol"
                  data-column-id="2"
                >
                  <div
                    class="sc-crXcEl gJhVev rdt_TableCol_Sortable"
                    data-column-id="2"
                    data-sort-id="2"
                    role="columnheader"
                    tabindex="0"
                  >
                    <div
                      class="sc-evZas eUeqdG"
                      data-column-id="2"
                    >
                      Role
                    </div>
                    <span
                      class="sc-kDDrLX fTxtMi"
                    >
                      ▲
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="sc-hHLeRK gFYXSL rdt_TableBody"
              role="rowgroup"
            >
              <div
                class="sc-jqUVSM gOWJSc rdt_TableRow"
                id="row-0"
                role="row"
              >
                <div
                  class="sc-hKMtZM sc-eCYdqJ sc-jSMfEi ebHJoU bnAwAJ bMgaAx rdt_TableCell"
                  data-column-id="1"
                  data-tag="allowRowEvents"
                  id="cell-1-undefined"
                  role="gridcell"
                >
                  <div
                    data-tag="allowRowEvents"
                  >
                    Squeak Trainywhel
                  </div>
                </div>
                <div
                  class="sc-hKMtZM sc-eCYdqJ sc-jSMfEi ebHJoU bnAwAJ bMgaAx rdt_TableCell"
                  data-column-id="2"
                  data-tag="allowRowEvents"
                  id="cell-2-undefined"
                  role="gridcell"
                >
                  <div
                    data-tag="allowRowEvents"
                  >
                    Admin
                  </div>
                </div>
              </div>
              <div
                class="sc-jqUVSM gOWJSc rdt_TableRow"
                id="row-1"
                role="row"
              >
                <div
                  class="sc-hKMtZM sc-eCYdqJ sc-jSMfEi ebHJoU bnAwAJ bMgaAx rdt_TableCell"
                  data-column-id="1"
                  data-tag="allowRowEvents"
                  id="cell-1-undefined"
                  role="gridcell"
                >
                  <div
                    data-tag="allowRowEvents"
                  >
                    Isobel Jennery
                  </div>
                </div>
                <div
                  class="sc-hKMtZM sc-eCYdqJ sc-jSMfEi ebHJoU bnAwAJ bMgaAx rdt_TableCell"
                  data-column-id="2"
                  data-tag="allowRowEvents"
                  id="cell-2-undefined"
                  role="gridcell"
                >
                  <div
                    data-tag="allowRowEvents"
                  >
                    Membership Admin
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <nav
          class="sc-iIPllB jxflYm rdt_Pagination"
        >
          <span
            class="sc-bZkfAO sc-ikZpkk lmGpws kBhHpT"
          >
            Rows per page:
          </span>
          <div
            class="sc-llJcti gnaqej"
          >
            <select
              aria-label="Rows per page:"
              class="sc-cxabCf kJrhuj"
            >
              <option
                selected=""
                value="10"
              >
                10
              </option>
              <option
                value="15"
              >
                15
              </option>
              <option
                value="20"
              >
                20
              </option>
              <option
                value="25"
              >
                25
              </option>
              <option
                value="30"
              >
                30
              </option>
            </select>
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </div>
          <span
            class="sc-bZkfAO sc-kLLXSd lmGpws OHwJB"
          >
            1-2 of 2
          </span>
          <div
            class="sc-ezWOiH jMMJgk"
          >
            <button
              aria-disabled="true"
              aria-label="First Page"
              class="sc-gicCDI bvxQGL"
              disabled=""
              id="pagination-first-page"
              type="button"
            >
              <svg
                aria-hidden="true"
                height="24"
                role="presentation"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                />
                <path
                  d="M24 24H0V0h24v24z"
                  fill="none"
                />
              </svg>
            </button>
            <button
              aria-disabled="true"
              aria-label="Previous Page"
              class="sc-gicCDI bvxQGL"
              disabled=""
              id="pagination-previous-page"
              type="button"
            >
              <svg
                aria-hidden="true"
                height="24"
                role="presentation"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
            <button
              aria-disabled="true"
              aria-label="Next Page"
              class="sc-gicCDI bvxQGL"
              disabled=""
              id="pagination-next-page"
              type="button"
            >
              <svg
                aria-hidden="true"
                height="24"
                role="presentation"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
            <button
              aria-disabled="true"
              aria-label="Last Page"
              class="sc-gicCDI bvxQGL"
              disabled=""
              id="pagination-last-page"
              type="button"
            >
              <svg
                aria-hidden="true"
                height="24"
                role="presentation"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                />
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
              </svg>
            </button>
          </div>
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
